### PR TITLE
Functions hometask done.

### PR DESCRIPTION
**Pull request by kusem**

### CODE REVIEW

File: src/functions/Functions.php [hunk 1]
```diff
@@ -87,10 +91,14 @@ public function countArguments(): array
      * @see https://www.php.net/manual/en/migration56.new-features.php#migration56.new-features.splat
      *
      * @return array
-     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
      */
-    public function countArgumentsWrapper(): array
+    public function countArgumentsWrapper(...$arguments): array
```
Comment (yderevianko-cartera) [inline src/functions/Functions.php]: you should`t change function defenition

File: tests/CountArgumentsTest.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        $this->functions = new functions\Functions();
+    }
+
+    public function testPositiveString()
+    {
+        $this->assertEquals(
+            [
+                'argument_count' => 1,
+                'argument_values' => Array (0 => '#russiaisaterroriststate'),
```
Comment (yderevianko-cartera) [inline tests/CountArgumentsTest.php]: please use [] instead of Array

File: tests/CountArgumentsTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Tests if empty argument sent
+     */
+    public function testPositiveEmpty()
+    {
+        $this->assertEquals(
+            [
+                'argument_count' => 0,
+                'argument_values' => Array (),
```
Comment (yderevianko-cartera) [inline tests/CountArgumentsTest.php]: please use [] instead of Array

File: tests/CountArgumentsWrapperTest.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        $this->functions = new functions\Functions();
+    }
+
+    public function testNegative()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        $this->functions->countArgumentsWrapper(2, 4, 0, 2, ['russiaisaterroriststate']);
+    }
+
```
Comment (yderevianko-cartera) [inline tests/CountArgumentsWrapperTest.php]: not needed empty line

File: tests/SayHelloArgumentTest.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    public function positiveDataProvider(): array
+    {
+        return [
+            ['Epam', 'Hello Epam'],
+            [1, 'Hello 1'],
+            [true, 'Hello 1'],
+        ];
+    }
+}
```
Comment (yderevianko-cartera) [inline tests/SayHelloArgumentTest.php]: please add empty line at the end of file

File: tests/SayHelloArgumentWrapperTest.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        return [
+            'null'      => [null, InvalidArgumentException::class],
+            'double'    => [3.14, InvalidArgumentException::class],
+            'array'     => [['russia', 'is', 'a', 'terrorist', 'state'], InvalidArgumentException::class],
+            'stdClass'  => [new stdClass(), InvalidArgumentException::class],
+            //'string'  => ["", InvalidArgumentException::class], // should fail - valid data type, but exception is expected
+        ];
+    }
+}
```
Comment (yderevianko-cartera) [inline tests/SayHelloArgumentWrapperTest.php:34]: please add empty line at the end of file

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/functions/Functions.php (hunk 1 above)
Comment (kusem) [reply src/functions/Functions.php]: > you should`t change function defenition

All changes done

File: tests/CountArgumentsTest.php (hunk 2 above)
Comment (kusem) [reply tests/CountArgumentsTest.php]: > please use [] instead of Array

All changes done

File: tests/CountArgumentsTest.php (hunk 3 above)
Comment (kusem) [reply tests/CountArgumentsTest.php]: > please use [] instead of Array

All changes done

File: tests/CountArgumentsWrapperTest.php (hunk 4 above)
Comment (kusem) [reply tests/CountArgumentsWrapperTest.php]: > not needed empty line

All changes done

File: tests/SayHelloArgumentWrapperTest.php (hunk 6 above)
Comment (kusem) [reply tests/SayHelloArgumentWrapperTest.php:34]: > please add empty line at the end of file

All changes done

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**